scipy/stats/stats.py

```diff
@@ -5718,3 +5721,195 @@ def rankdata(a, method='average'):
 
     # average method
     return .5 * (count[dense] + count[dense - 1] + 1)
+
+
+def rvs_ratio_unif(pdf, size=1, x1=None, x2=None, y2=None, shift=0,
+                   maxiter=None, bounds=None, random_state=None):
+    """
+    Generate random samples from a probability density function using the
+    ratio-of-uniforms method.
+
+    Parameters
+    ----------
+    pdf : callable
+        The probability density function of the distribution.
+    size : int or tuple of ints, optional
+        Defining number of random variates (default is 1).
+    x1 : float, optional
+        The lower bound of the bounding rectangle in the x-direction.
+        Default is None. In that case, attempt to find bound numerically.
+    x2 : float, optional
+        The upper bound of the bounding rectangle in the x-direction.
+        Default is None. In that case, attempt to find bound numerically.
+    y2 : float, optional
+        The upper bound of the bounding rectangle in the y-direction.
+        Default is None. In that case, attempt to find bound numerically.
+    shift : float, optional.
+        Shift parameter of ratio-of-uniforms method, see Notes. Default is 0.
+    maxiter : int, optional
+        Maximum number of iterations to generate the random variables.
+        Default is None. In that case, maxiter is set to a value that
+        ensures successful simulation with a high probability.
+    bounds : tuple of floats (a, b), optional
+        Default is None. In case any of the values x1, x2, y2 are not
+        specified (i.e. equal to None), attempt to find the value numerically
+        over the interval from a to b.
+    random_state : None or int or np.random.RandomState instance, optional
+        If int or RandomState, use it for drawing the random variates.
+        If None, use np.random.RandomState. Default is None.
+
+    Returns
+    -------
+    rvs : array
+        the random variates distributed according to the probability
+        distrubtion, pdf
+
+    Notes
+    -----
+    Given a univariate probability density function (pdf) f and a constant c,
+    define the set A(c) = {(x, y) : 0 < y <= sqrt(f(x/y + c)) }.
+    If (U, V) is a random vector uniformly distributed over A(c), then U/V + c
+    follows a distribution with pdf f.
+
+    The above result (see [1]_, [2]_) can be used to sample random variables
+    using only the pdf, i.e. no inversion of the cdf is required. Typical
+    choices of c are zero or the mode of f. The set A(c) is a subset of the
+    rectangle R = [-x1, x2] x [0, y2] where
+    - x1 = inf (x - c) sqrt(f(x))
```

Comment:
I think this needs a blank line above it to render as list. And again double backticks.

Reply:
There are many other formatting issues. Here is the rendered docstring: https://5360-1460385-gh.circle-artifacts.com/0/home/ubuntu/scipy/doc/build/html-scipyorg/generated/scipy.stats.rvs_ratio_unif.html#scipy.stats.rvs_ratio_unif (you may need to log in with your GitHub account).

Reply:
Where can I find these links?